web/ts/chat/Poll.ts

```diff
@@ -0,0 +1,64 @@
+import { startPoll } from "../watch";
+
+export class Poll {
+    readonly streamId: number;
+
+    activePoll: object;
+    result: object;
+    showCreateUI: boolean;
+    question: string;
+    options: object[];
+
+    constructor(streamId: number) {
+        this.streamId = streamId;
+        this.activePoll = null;
+        this.result = null;
+        this.reset();
+    }
+
+    start() {
+        startPoll(
+            this.question,
+            // @ts-ignore
+            this.options.map(({ answer }) => answer),
+        );
+        this.reset();
+    }
+
+    async load() {
+        await fetch("/api/chat/" + this.streamId + "/active-poll")
```

Comment:
Again just something really minor but probably its simpler to stick to await:
```
const res = await fetch("/api/chat/" + this.streamId + "/active-poll");
this.activePoll = await res.json();
```
Just wanted to mention it, but I am also cool with the current approach :)